python/tach/parsing/modules.py

```diff
@@ -22,12 +22,51 @@ def find_duplicate_modules(modules: list[ModuleConfig]) -> list[str]:
     return duplicate_module_paths
 
 
-def build_module_tree(source_root: Path, modules: list[ModuleConfig]) -> ModuleTree:
+def find_cycle(
```

Comment:
Could you add a few test cases for this function?

Reply:
I believe the tests should go in `test_parsing.py`